docs/content/guide/i18n.ngdoc

```diff
@@ -142,84 +144,179 @@ displaying the date with a timezone specified by the developer.
-default `{{` and `}}` are allowed.
-
-This syntax extension, while based on MessageFormat, has
-been designed to be backwards compatible with existing
-AngularJS interpolation expressions.  The key rule is simply
-this: **All interpolations are done inside double curlies.**
-The top level comma operator after an expression inside the
-double curlies causes MessageFormat extensions to be
-recognized.  Such a top level comma is otherwise illegal in
-an Angular expression and is used by MessageFormat to
-specify the function (such as plural/select) and it's
-related syntax.
-
-To understand the extension, take a look at the ICU
-MessageFormat syntax as specified by the ICU documentation.
-Anywhere in that MessageFormat that you have regular message
-text and you want to substitute an expression, just put it
-in double curlies instead of single curlies that
-MessageFormat dictates.  This has a huge advantage.  **You
-are no longer limited to simple identifiers for
-substitutions**.  Because you are using double curlies, you
-can stick in any arbitrary interpolation syntax there,
-including nesting more MessageFormat expressions!  Some
-examples will make this clear.  In the following example, I
-will only be showing you the AngularJS syntax.
+#### Messages
 
+Messages immediately follow a selection keyword and are optionally preceded by whitespace.  They are
+written in single curly braces (`{}`).  They may contain Angular interpolation syntax inside them.
+In addition, the `#` symbol is a placeholder for the actual numeric value of the expression.
 
 ### Simple plural example
 
-```
+```text
 {{numMessages, plural,
-      =0 { You have no new messages }
-      =1 { You have one new message }
-   other { You have # new messages }
+      =0 {You have no new messages}
+      =1 {You have one new message}
+   other {You have # new messages}
 }}
 ```
 
-While I won't be teaching you MessageFormat here, you will
-note that the `#` symbol works as expected.  You could have
-also written it as:
+Because these messages can themselves contain Angular expressions, you could also write this as
+follows:
 
-```
+```text
 {{numMessages, plural,
-      =0 { You have no new messages }
-      =1 { You have one new message }
-   other { You have {{numMessages}} new messages }
+      =0 {You have no new messages}
+      =1 {You have one new message}
+   other {You have {{numMessages}} new messages}
 }}
 ```
 
-where you explicitly typed in `numMessages` for "other"
-instead of using `#`.  They are nearly the same except if
-you're using "offset".  Refer to the ICU MessageFormat
-documentation to learn about offset.
 
-Please note that **other** is a **required** category (for
-both the plural syntax and the select syntax that is shown
-later.)
+### Plural syntax with optional `offset`
+
+The plural syntax supports an optional `offset` syntax that is used in matching.  It's simpler to
+explain this with an example.
+
+```text
+{{recipients.length, plural, offset:1
```

Comment:
For pluralization, negative numbers don't make sense—it's a logic bug.  If you're using an offset, your messages are automatically written to use this information (e.g. "… and # other people…").  You can simply add exact matches for the cases that would cause you to have negative "#" evaluations.